File: frontend/lib/models/cuisine.dart
class Cuisine {
  const Cuisine({
    required this.id,
    required this.name,
    this.imageUrl,
  });

  factory Cuisine.fromJson(Map<String, dynamic> json) {
    return Cuisine(
      id: json['_id'] as String,
      name: json['name'] as String? ?? '',
      imageUrl: json['imageUrl'] as String?,
    );
  }

  final String id;
  final String name;
  final String? imageUrl;
}

File: backend/src/models/Cuisine.js
const mongoose = require('mongoose');

const cuisineSchema = new mongoose.Schema(
  {
    name: { type: String, required: true, unique: true },
    imageUrl: { type: String, default: '' },
  },
  { timestamps: true }
);

module.exports = mongoose.model('Cuisine', cuisineSchema);

